src/components/primitives/Forms/IconRight.tsx

```diff
@@ -0,0 +1,31 @@
+import { FiUnlock, FiX } from 'react-icons/fi';
+import { ClickableIcon } from './AddressInput';
+
+export const IconRight = ({
+  disabled,
+  value,
+  onChange,
+  defaultValue,
+  setDisabled,
+  type,
+}) => {
+  if (defaultValue) {
+    return (
+      <ClickableIcon
+        aria-label="enable"
+        onClick={() => {
+          setDisabled(false);
+        }}
+      >
+        <FiUnlock size={18} />
+      </ClickableIcon>
+    );

```

Comment:
This causes the lock button to have a one-way toggle. If it has a `defaultValue`, it is locked, and when clicking it, unlocks the field.

But the icon doesn't disappear, and further clicking doesn't lock the field again, but does nothing because `onClick={()=> setDisabled(false)}`

Reply:
i added a validation to show the unlock icon only if the input is disabled, which makes sense